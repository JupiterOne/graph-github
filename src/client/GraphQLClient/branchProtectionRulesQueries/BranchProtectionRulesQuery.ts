import {
  BaseQueryState,
  BranchProtectionRuleResponse,
  BuildQuery,
  ProcessResponse,
} from '../types';
import { ExecutableQuery } from '../CreateQueryExecutor';
import fragments from '../fragments';
import { MAX_REQUESTS_LIMIT } from '../paginate';
import utils from '../utils';
import {
  branchProtectionRuleFields,
  buildVersionSafeFragments,
  processActors,
} from './shared';

interface QueryState extends BaseQueryState {
  isInitialQuery?: boolean;
}

export type QueryParams = {
  repoName: string;
  repoOwner: string;
  gheServerVersion?: string;
};

const buildQuery: BuildQuery<QueryParams, QueryState> = (
  queryParams,
  queryState,
): ExecutableQuery => {
  const versionSafeFragments = buildVersionSafeFragments(
    queryParams.gheServerVersion,
  );

  const query = `
      query (
        $repoName: String!
        $repoOwner: String!
        $maxLimit: Int!
      ) {
          repository(name: $repoName, owner: $repoOwner) {
            id
            name
            branchProtectionRules(first: $maxLimit) {
              nodes {
                ${branchProtectionRuleFields(versionSafeFragments)}
              }
            }
          }
          ...${fragments.rateLimit}
      }
  `;

  return {
    query,
    ...(queryState?.rateLimit && {
      rateLimit: queryState.rateLimit,
    }),
    queryVariables: {
      repoName: queryParams.repoName,
      repoOwner: queryParams.repoOwner,
      maxLimit: MAX_REQUESTS_LIMIT,
    },
  };
};

const processResponseData: ProcessResponse<
  BranchProtectionRuleResponse,
  QueryState
> = async (responseData, iteratee) => {
  const rateLimit = responseData.rateLimit;
  const rules = responseData.repository?.branchProtectionRules.nodes ?? [];

  for (const rule of rules) {
    if (!utils.hasProperties(rule)) {
      continue;
    }

    const processedRule = {
      repoId: responseData.repository.id,
      repoName: responseData.repository.name,
      ...rule,
      bypassForcePushAllowances: processActors(
        rule.bypassForcePushAllowances?.nodes,
      ),
      bypassPullRequestAllowances: processActors(
        rule.bypassPullRequestAllowances?.nodes,
      ),
      pushAllowances: processActors(rule.pushAllowances?.nodes),
      reviewDismissalAllowances: processActors(
        rule.reviewDismissalAllowances?.nodes,
      ),
    };

    await iteratee(processedRule);
  }

  return {
    rateLimit,
  };
};

const iterateBranchProtectionRules = async (
  queryParams: QueryParams,
  execute,
  iteratee,
) => {
  let queryState: QueryState = {};
  const executable = buildQuery(queryParams, queryState);
  const response = await execute(executable);
  queryState = await processResponseData(response, iteratee);

  const queryCost = queryState?.rateLimit?.cost ?? 0;

  return {
    totalCost: queryCost,
    limit: queryState?.rateLimit?.limit,
    remaining: queryState?.rateLimit?.remaining,
    resetAt: queryState?.rateLimit?.resetAt,
  };
};

export default { iterateBranchProtectionRules };
