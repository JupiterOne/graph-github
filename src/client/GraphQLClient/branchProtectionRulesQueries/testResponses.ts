const branchProtectionRulesResponses = [
  {
    repository: {
      repoId: 'R_kgDOI2pL-A',
      name: 'reimagined-barnacle',
      branchProtectionRules: {
        nodes: [
          {
            id: 'BPR_sdflk_slkwefoi',
            requiresLinearHistory: false,
            requiredApprovingReviewCount: 2,
            dismissesStaleReviews: false,
            requiresCodeOwnerReviews: false,
            requiresCommitSignatures: true,
            isAdminEnforced: false,
            allowsForcePushes: false,
            allowsDeletions: false,
            blocksCreations: false,
            requiresConversationResolution: false,
            pattern: 'release-*',
            requiresApprovingReviews: true,
            requiredStatusCheckContexts: [],
            creator: {
              login: 'VDubber',
            },
            databaseId: 25221868,
            requiresStatusChecks: false,
            requiresStrictStatusChecks: true,
            restrictsPushes: false,
            restrictsReviewDismissals: false,
            requiredStatusChecks: [],
            bypassPullRequestAllowances: {
              nodes: [
                {
                  actor: {
                    __typename: 'Team',
                    id: 'T_kwDOBfl5Zc4AWFta',
                    name: 'j1-nested-team',
                  },
                },
                {
                  actor: {
                    __typename: 'User',
                    id: 'MDQ6VXNlcjU1NzcxODIz',
                    login: 'adam-in-ict',
                    email: '',
                  },
                },
              ],
            },
          },
          {
            requiresLinearHistory: false,
            requiredApprovingReviewCount: 1,
            dismissesStaleReviews: true,
            requiresCodeOwnerReviews: false,
            requiresCommitSignatures: false,
            isAdminEnforced: true,
            allowsForcePushes: false,
            allowsDeletions: false,
            blocksCreations: false,
            requiresConversationResolution: false,
            pattern: 'main',
            requiresApprovingReviews: true,
            requiredStatusCheckContexts: [],
            creator: {
              login: 'electricgull',
            },
            databaseId: 27924631,
            requiresStatusChecks: true,
            requiresStrictStatusChecks: false,
            restrictsPushes: false,
            restrictsReviewDismissals: false,
            requiredStatusChecks: [],
            bypassPullRequestAllowances: {
              nodes: [
                {
                  actor: {
                    __typename: 'Team',
                    id: 'T_kwDOBfl5Zc4AWFta',
                    name: 'j1-nested-team',
                  },
                },
                {
                  actor: {
                    __typename: 'User',
                    id: 'MDQ6VXNlcjE2NTYyMDg1',
                    login: 'electricgull',
                    email: '',
                  },
                },
              ],
            },
          },
        ],
      },
    },
  },
];

export { branchProtectionRulesResponses };
