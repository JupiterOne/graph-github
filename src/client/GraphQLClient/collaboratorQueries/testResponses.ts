const repoCollaborators = [
  {
    repository: {
      id: 'R_kgDOG_MJdQ',
      collaborators: {
        edges: [
          {
            node: {
              id: 'MDQ6VXNlcjU5MjMwMDI=',
              name: 'Samuel Poulton',
              login: 'VDubber',
            },
            permission: 'ADMIN',
          },
          {
            node: {
              id: 'U_kgDOBgq4WA',
              name: null,
              login: 'spoultonV2',
            },
            permission: 'READ',
          },
        ],
        pageInfo: {
          endCursor: 'more2come==',
          hasNextPage: true,
        },
      },
    },
    rateLimit: {
      cost: 2,
    },
  },
  {
    repository: {
      id: 'R_kgDOG_MJdQ',
      collaborators: {
        edges: [
          {
            node: {
              id: 'MDQ6VXNlcjU5MjMwMDI=',
              name: 'Harper Smith',
              login: 'HSM',
            },
            permission: 'ADMIN',
          },
          {
            node: {
              id: 'U_kgDOBgq4WA',
              name: 'Unkonw 2',
              login: 'u2',
            },
            permission: 'READ',
          },
        ],
        pageInfo: {
          endCursor: 'more2come==',
          hasNextPage: false,
        },
      },
    },
    rateLimit: {
      cost: 4,
    },
  },
];

export { repoCollaborators };
