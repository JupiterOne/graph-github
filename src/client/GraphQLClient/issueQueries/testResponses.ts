const issuesResponses = [
  {
    search: {
      issueCount: 3,
      edges: [
        {
          node: {
            id: '111',
            title: 'Need to add code part 1',
            assignees: {
              totalCount: 1,
              nodes: [
                {
                  name: null,
                  login: 'spoultonV2',
                },
              ],
              pageInfo: {
                endCursor: 'Y3Vyc29yOnYyOpHOBgq4WA==',
                hasNextPage: false,
              },
            },
            labels: {
              totalCount: 9,
              nodes: [
                {
                  id: 'LA_kwDOG5ypUc7mjrvd',
                  name: 'bug',
                },
                {
                  id: 'LA_kwDOG5ypUc7mjrvi',
                  name: 'feature',
                },
              ],
              pageInfo: {
                endCursor:
                  'Y3Vyc29yOnYyOpK5MjAyMi0wMi0yNFQxMTowNDo1OC0wNzowMM7mjrwE',
                hasNextPage: false,
              },
            },
          },
        },
        {
          node: {
            id: '2222',
            title: 'Need to add code part 2',
            assignees: {
              totalCount: 1,
              nodes: [
                {
                  name: null,
                  login: 'maxmin',
                },
              ],
              pageInfo: {
                endCursor: 'Y3Vyc29yOnYyOpHOBgq4WA==',
                hasNextPage: false,
              },
            },
            labels: {
              totalCount: 9,
              nodes: [
                {
                  id: 'LA_kwDOG5ypUc7mjrvd',
                  name: 'bug',
                },
                {
                  id: 'LA_kwDOG5ypUc7mjrvi',
                  name: 'critical',
                },
              ],
              pageInfo: {
                endCursor:
                  'Y3Vyc29yOnYyOpK5MjAyMi0wMi0yNFQxMTowNDo1OC0wNzowMM7mjrwE',
                hasNextPage: false,
              },
            },
          },
        },
      ],
      pageInfo: {
        endCursor: 'Y3Vyc29yOjE=',
        hasNextPage: true,
      },
    },
  },
  // Send page
  {
    search: {
      issueCount: 3,
      edges: [
        {
          node: {
            id: '333',
            title: 'Need to add code part 3',
            assignees: {
              totalCount: 1,
              nodes: [
                {
                  name: null,
                  login: 'chris3',
                },
              ],
              pageInfo: {
                endCursor: 'Y3Vyc29yOnYyOpHOBgq4WA==',
                hasNextPage: false,
              },
            },
            labels: {
              totalCount: 9,
              nodes: [
                {
                  id: 'LA_kwDOG5ypUc7mjrvd',
                  name: 'defect',
                },
                {
                  id: 'LA_kwDOG5ypUc7mjrvi',
                  name: 'easy',
                },
              ],
              pageInfo: {
                endCursor:
                  'Y3Vyc29yOnYyOpK5MjAyMi0wMi0yNFQxMTowNDo1OC0wNzowMM7mjrwE',
                hasNextPage: false,
              },
            },
          },
        },
      ],
      pageInfo: {
        endCursor: 'Y3Vyc29yOjE=',
        hasNextPage: false,
      },
    },
    rateLimit: {
      cost: 3,
      remaining: 4983,
    },
  },
];

export { issuesResponses };
