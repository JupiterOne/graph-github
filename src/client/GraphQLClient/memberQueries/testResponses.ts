const orgMembers = [
  {
    organization: {
      id: 'O_kgDOBfl5ZQ',
      membersWithRole: {
        totalCount: 4,
        edges: [
          {
            node: {
              id: 'm_1',
              login: 'VDubber',
              name: 'Samuel Poulton',
              isSiteAdmin: false,
              company: 'JupiterOne',
              createdAt: '2013-11-12T20:20:50Z',
              databaseId: 5923002,
              email: 'samuel.poulton@jupiterone.com',
              isEmployee: false,
              location: null,
              updatedAt: '2022-03-15T15:43:54Z',
              url: 'https://github.com/VDubber',
              websiteUrl: null,
            },
            hasTwoFactorEnabled: true,
            role: 'ADMIN',
          },
          {
            node: {
              id: 'm_2',
              login: 'codestuffs',
              name: 'Jump Jack',
              isSiteAdmin: false,
              company: null,
              createdAt: '2022-03-10T21:30:13Z',
              databaseId: 101365848,
              email: 'jump@jack.com',
              isEmployee: false,
              location: null,
              updatedAt: '2022-03-10T21:30:13Z',
              url: 'https://github.com/codestuffs',
              websiteUrl: null,
            },
            hasTwoFactorEnabled: false,
            role: 'MEMBER',
          },
        ],
        pageInfo: {
          endCursor: 'more2come',
          hasNextPage: true,
        },
      },
    },
    rateLimit: {
      cost: 5,
    },
  },
  {
    organization: {
      id: 'O_kgDOBfl5ZQ',
      membersWithRole: {
        totalCount: 4,
        edges: [
          {
            node: {
              id: 'm_3',
              login: 'JP',
              name: 'Joe Poulton',
              isSiteAdmin: false,
              company: 'JupiterOne',
              createdAt: '2013-11-12T20:20:50Z',
              databaseId: 5923002,
              email: 'j.poulton@jupiterone.com',
              isEmployee: false,
              location: null,
              updatedAt: '2022-03-15T15:43:54Z',
              url: 'https://github.com/JP',
              websiteUrl: null,
            },
            hasTwoFactorEnabled: true,
            role: 'ADMIN',
          },
          {
            node: {
              id: 'm_4',
              login: 'spv2',
              name: null,
              isSiteAdmin: false,
              company: null,
              createdAt: '2022-03-10T21:30:13Z',
              databaseId: 101365848,
              email: 'spv2@j1.com',
              isEmployee: false,
              location: null,
              updatedAt: '2022-03-10T21:30:13Z',
              url: 'https://github.com/spv2',
              websiteUrl: null,
            },
            hasTwoFactorEnabled: false,
            role: 'MEMBER',
          },
        ],
        pageInfo: {
          endCursor: 'end==',
          hasNextPage: false,
        },
      },
    },
    rateLimit: {
      cost: 5,
    },
  },
];

const teamMembers = [
  {
    organization: {
      id: 'O_kgDOBfl5ZQ',
      team: {
        id: 'T_kwDOBfl5Zc4AWFtb',
        name: 'j1-ingest-eng',
        members: {
          edges: [
            {
              node: {
                id: 'm1',
                name: 'Joe Poulton',
                login: 'JP',
              },
              role: 'MAINTAINER',
            },
            {
              node: {
                id: 'm2',
                name: 'Samuel Poulton',
                login: 'VDubber',
              },
              role: 'ADMIN',
            },
            {
              node: {
                id: 'm3',
                name: 'E',
                login: 'EP',
              },
              role: 'MAINTAINER',
            },
          ],
          pageInfo: {
            endCursor: 'more2come',
            hasNextPage: true,
          },
        },
      },
    },
    rateLimit: {
      cost: 2,
    },
  },
  {
    organization: {
      id: 'O_kgDOBfl5ZQ',
      team: {
        id: 'T_kwDOBfl5Zc4AWFtb',
        name: 'j1-ingest-eng',
        members: {
          edges: [
            {
              node: {
                id: 'm4',
                name: 'Silvia Smith',
                login: 'SSM',
              },
              role: 'MAINTAINER',
            },
          ],
          pageInfo: {
            endCursor: 'end',
            hasNextPage: false,
          },
        },
      },
    },
    rateLimit: {
      cost: 1,
    },
  },
];

export { orgMembers, teamMembers };
