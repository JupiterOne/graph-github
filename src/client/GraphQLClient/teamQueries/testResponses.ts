const teams = [
  {
    organization: {
      id: 'O_kgDOBfl5ZQ',
      teams: {
        edges: [
          {
            node: {
              id: 'T_kwDOBfl5Zc4AWHaA',
              name: 'j1-empty-team',
              url: 'https://github.com/orgs/j1-ingest/teams/j1-empty-team',
              slug: 'j1-empty-team',
              createdAt: '2022-03-14T16:40:24Z',
              updatedAt: '2022-03-14T16:40:24Z',
              databaseId: 5797504,
              description: '',
              privacy: 'VISIBLE',
            },
          },
          {
            node: {
              id: 'T_kwDOBfl5Zc4AV-Cl',
              name: 'j1-github-ingest',
              url: 'https://github.com/orgs/j1-ingest/teams/j1-github-ingest',
              slug: 'j1-github-ingest',
              createdAt: '2022-03-04T21:51:41Z',
              updatedAt: '2022-03-04T21:51:41Z',
              databaseId: 5759141,
              description: 'Testing the GitHub Integration',
              privacy: 'VISIBLE',
            },
          },
        ],
        pageInfo: {
          endCursor: 'more2come',
          hasNextPage: true,
        },
      },
    },
    rateLimit: {
      cost: 2,
    },
  },
  {
    organization: {
      id: 'O_kgDOBfl5ZQ',
      teams: {
        edges: [
          {
            node: {
              id: 'T_kwDOBfl5Zc4AWFtb',
              name: 'j1-ingest-eng',
              url: 'https://github.com/orgs/j1-ingest/teams/j1-ingest-eng',
              slug: 'j1-ingest-eng',
              createdAt: '2022-03-11T22:03:54Z',
              updatedAt: '2022-03-11T22:03:54Z',
              databaseId: 5790555,
              description: 'Test team',
              privacy: 'SECRET',
            },
          },
          {
            node: {
              id: 'T_kwDOBfl5Zc4AWFta',
              name: 'j1-nested-team',
              url: 'https://github.com/orgs/j1-ingest/teams/j1-nested-team',
              slug: 'j1-nested-team',
              createdAt: '2022-03-11T22:03:02Z',
              updatedAt: '2022-03-11T22:03:02Z',
              databaseId: 5790554,
              description: 'Is this team nested?',
              privacy: 'VISIBLE',
            },
          },
        ],
        pageInfo: {
          endCursor: 'end==',
          hasNextPage: false,
        },
      },
    },
    rateLimit: {
      cost: 1,
    },
  },
];

export { teams };
