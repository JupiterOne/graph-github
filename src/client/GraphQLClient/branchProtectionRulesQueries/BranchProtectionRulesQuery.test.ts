import BranchProtectionRulesQuery from './BranchProtectionRulesQuery';
import { branchProtectionRulesResponses } from './testResponses';
import { validate } from '@octokit/graphql-schema';

describe('BranchProtectionRulesQuery', () => {
  describe('#iterateBranchProtectionRules', () => {
    test('Pulling data out', async () => {
      const mockGheVersions = ['3.4.0', '3.5.0', '3.6.0'];

      for (const version of mockGheVersions) {
        const execute = jest
          .fn()
          .mockResolvedValue(branchProtectionRulesResponses[0]);

        const iteratee = jest.fn();

        // Act
        const result =
          await BranchProtectionRulesQuery.iterateBranchProtectionRules(
            {
              repoOwner: 'J1-Test',
              repoName: 'happy-sunshine',
              gheServerVersion: version,
            },
            execute,
            iteratee,
          );

        // Assert
        expect(result.totalCost).toBe(0);
        expect(iteratee).toHaveBeenCalledTimes(2);
        expect(iteratee.mock.calls[0][0]).toMatchSnapshot();
        expect(iteratee.mock.calls[1][0]).toMatchSnapshot();
        expect(execute).toHaveBeenCalledTimes(1);
        expect(execute.mock.calls[0][0]).toHaveProperty('query');
        expect(() => validate(execute.mock.calls[0][0].query)).not.toThrow();
        expect(execute.mock.calls[0][0]).toMatchSnapshot();
      }
    });
  });
});
