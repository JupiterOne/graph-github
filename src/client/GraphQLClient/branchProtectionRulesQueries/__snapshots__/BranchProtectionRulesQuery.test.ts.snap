// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BranchProtectionRulesQuery #iterateBranchProtectionRules Pulling data out 1`] = `
Object {
  "allowsDeletions": false,
  "allowsForcePushes": false,
  "blocksCreations": false,
  "bypassForcePushAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
  "bypassPullRequestAllowances": Object {
    "apps": Array [],
    "teams": Array [
      Object {
        "__typename": "Team",
        "id": "T_kwDOBfl5Zc4AWFta",
        "name": "j1-nested-team",
      },
    ],
    "users": Array [
      Object {
        "__typename": "User",
        "email": "",
        "id": "MDQ6VXNlcjU1NzcxODIz",
        "login": "adam-in-ict",
      },
    ],
  },
  "creator": Object {
    "login": "VDubber",
  },
  "databaseId": 25221868,
  "dismissesStaleReviews": false,
  "id": "BPR_sdflk_slkwefoi",
  "isAdminEnforced": false,
  "pattern": "release-*",
  "pushAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
  "repoId": undefined,
  "repoName": "reimagined-barnacle",
  "requiredApprovingReviewCount": 2,
  "requiredStatusCheckContexts": Array [],
  "requiredStatusChecks": Array [],
  "requiresApprovingReviews": true,
  "requiresCodeOwnerReviews": false,
  "requiresCommitSignatures": true,
  "requiresConversationResolution": false,
  "requiresLinearHistory": false,
  "requiresStatusChecks": false,
  "requiresStrictStatusChecks": true,
  "restrictsPushes": false,
  "restrictsReviewDismissals": false,
  "reviewDismissalAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
}
`;

exports[`BranchProtectionRulesQuery #iterateBranchProtectionRules Pulling data out 2`] = `
Object {
  "allowsDeletions": false,
  "allowsForcePushes": false,
  "blocksCreations": false,
  "bypassForcePushAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
  "bypassPullRequestAllowances": Object {
    "apps": Array [],
    "teams": Array [
      Object {
        "__typename": "Team",
        "id": "T_kwDOBfl5Zc4AWFta",
        "name": "j1-nested-team",
      },
    ],
    "users": Array [
      Object {
        "__typename": "User",
        "email": "",
        "id": "MDQ6VXNlcjE2NTYyMDg1",
        "login": "electricgull",
      },
    ],
  },
  "creator": Object {
    "login": "electricgull",
  },
  "databaseId": 27924631,
  "dismissesStaleReviews": true,
  "isAdminEnforced": true,
  "pattern": "main",
  "pushAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
  "repoId": undefined,
  "repoName": "reimagined-barnacle",
  "requiredApprovingReviewCount": 1,
  "requiredStatusCheckContexts": Array [],
  "requiredStatusChecks": Array [],
  "requiresApprovingReviews": true,
  "requiresCodeOwnerReviews": false,
  "requiresCommitSignatures": false,
  "requiresConversationResolution": false,
  "requiresLinearHistory": false,
  "requiresStatusChecks": true,
  "requiresStrictStatusChecks": false,
  "restrictsPushes": false,
  "restrictsReviewDismissals": false,
  "reviewDismissalAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
}
`;

exports[`BranchProtectionRulesQuery #iterateBranchProtectionRules Pulling data out 3`] = `
Object {
  "query": "
      query (
        $repoName: String!
        $repoOwner: String!
        $maxLimit: Int!
      ) {
          repository(name: $repoName, owner: $repoOwner) {
            id
            name
            branchProtectionRules(first: $maxLimit) {
              nodes {
                
  id
  requiresLinearHistory
  requiredApprovingReviewCount
  dismissesStaleReviews
  requiresCodeOwnerReviews
  requiresCommitSignatures
  isAdminEnforced
  allowsForcePushes
  allowsDeletions
  requiresConversationResolution
  pattern
  allowsDeletions
  requiresApprovingReviews
  requiredStatusCheckContexts
  creator {
    login
  }
  databaseId
  requiresStatusChecks
  requiresStrictStatusChecks
  restrictsPushes
  restrictsReviewDismissals
  requiredStatusChecks {
    context
    app {
      id
      name
    }
  }
  
  bypassForcePushAllowances {
    totalCount
  }
  bypassPullRequestAllowances {
    totalCount
  }
  pushAllowances {
    totalCount
  }
  reviewDismissalAllowances {
    totalCount
  }

              }
            }
          }
          ...on Query {
    rateLimit {
      limit
      cost
      remaining
      resetAt
    }
  }
      }
  ",
  "queryVariables": Object {
    "maxLimit": 100,
    "repoName": "happy-sunshine",
    "repoOwner": "J1-Test",
  },
}
`;

exports[`BranchProtectionRulesQuery #iterateBranchProtectionRules Pulling data out 4`] = `
Object {
  "allowsDeletions": false,
  "allowsForcePushes": false,
  "blocksCreations": false,
  "bypassForcePushAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
  "bypassPullRequestAllowances": Object {
    "apps": Array [],
    "teams": Array [
      Object {
        "__typename": "Team",
        "id": "T_kwDOBfl5Zc4AWFta",
        "name": "j1-nested-team",
      },
    ],
    "users": Array [
      Object {
        "__typename": "User",
        "email": "",
        "id": "MDQ6VXNlcjU1NzcxODIz",
        "login": "adam-in-ict",
      },
    ],
  },
  "creator": Object {
    "login": "VDubber",
  },
  "databaseId": 25221868,
  "dismissesStaleReviews": false,
  "id": "BPR_sdflk_slkwefoi",
  "isAdminEnforced": false,
  "pattern": "release-*",
  "pushAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
  "repoId": undefined,
  "repoName": "reimagined-barnacle",
  "requiredApprovingReviewCount": 2,
  "requiredStatusCheckContexts": Array [],
  "requiredStatusChecks": Array [],
  "requiresApprovingReviews": true,
  "requiresCodeOwnerReviews": false,
  "requiresCommitSignatures": true,
  "requiresConversationResolution": false,
  "requiresLinearHistory": false,
  "requiresStatusChecks": false,
  "requiresStrictStatusChecks": true,
  "restrictsPushes": false,
  "restrictsReviewDismissals": false,
  "reviewDismissalAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
}
`;

exports[`BranchProtectionRulesQuery #iterateBranchProtectionRules Pulling data out 5`] = `
Object {
  "allowsDeletions": false,
  "allowsForcePushes": false,
  "blocksCreations": false,
  "bypassForcePushAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
  "bypassPullRequestAllowances": Object {
    "apps": Array [],
    "teams": Array [
      Object {
        "__typename": "Team",
        "id": "T_kwDOBfl5Zc4AWFta",
        "name": "j1-nested-team",
      },
    ],
    "users": Array [
      Object {
        "__typename": "User",
        "email": "",
        "id": "MDQ6VXNlcjE2NTYyMDg1",
        "login": "electricgull",
      },
    ],
  },
  "creator": Object {
    "login": "electricgull",
  },
  "databaseId": 27924631,
  "dismissesStaleReviews": true,
  "isAdminEnforced": true,
  "pattern": "main",
  "pushAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
  "repoId": undefined,
  "repoName": "reimagined-barnacle",
  "requiredApprovingReviewCount": 1,
  "requiredStatusCheckContexts": Array [],
  "requiredStatusChecks": Array [],
  "requiresApprovingReviews": true,
  "requiresCodeOwnerReviews": false,
  "requiresCommitSignatures": false,
  "requiresConversationResolution": false,
  "requiresLinearHistory": false,
  "requiresStatusChecks": true,
  "requiresStrictStatusChecks": false,
  "restrictsPushes": false,
  "restrictsReviewDismissals": false,
  "reviewDismissalAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
}
`;

exports[`BranchProtectionRulesQuery #iterateBranchProtectionRules Pulling data out 6`] = `
Object {
  "query": "
      query (
        $repoName: String!
        $repoOwner: String!
        $maxLimit: Int!
      ) {
          repository(name: $repoName, owner: $repoOwner) {
            id
            name
            branchProtectionRules(first: $maxLimit) {
              nodes {
                
  id
  requiresLinearHistory
  requiredApprovingReviewCount
  dismissesStaleReviews
  requiresCodeOwnerReviews
  requiresCommitSignatures
  isAdminEnforced
  allowsForcePushes
  allowsDeletions
  requiresConversationResolution
  pattern
  allowsDeletions
  requiresApprovingReviews
  requiredStatusCheckContexts
  creator {
    login
  }
  databaseId
  requiresStatusChecks
  requiresStrictStatusChecks
  restrictsPushes
  restrictsReviewDismissals
  requiredStatusChecks {
    context
    app {
      id
      name
    }
  }
  blocksCreations
  bypassForcePushAllowances {
    totalCount
  }
  bypassPullRequestAllowances {
    totalCount
  }
  pushAllowances {
    totalCount
  }
  reviewDismissalAllowances {
    totalCount
  }

              }
            }
          }
          ...on Query {
    rateLimit {
      limit
      cost
      remaining
      resetAt
    }
  }
      }
  ",
  "queryVariables": Object {
    "maxLimit": 100,
    "repoName": "happy-sunshine",
    "repoOwner": "J1-Test",
  },
}
`;

exports[`BranchProtectionRulesQuery #iterateBranchProtectionRules Pulling data out 7`] = `
Object {
  "allowsDeletions": false,
  "allowsForcePushes": false,
  "blocksCreations": false,
  "bypassForcePushAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
  "bypassPullRequestAllowances": Object {
    "apps": Array [],
    "teams": Array [
      Object {
        "__typename": "Team",
        "id": "T_kwDOBfl5Zc4AWFta",
        "name": "j1-nested-team",
      },
    ],
    "users": Array [
      Object {
        "__typename": "User",
        "email": "",
        "id": "MDQ6VXNlcjU1NzcxODIz",
        "login": "adam-in-ict",
      },
    ],
  },
  "creator": Object {
    "login": "VDubber",
  },
  "databaseId": 25221868,
  "dismissesStaleReviews": false,
  "id": "BPR_sdflk_slkwefoi",
  "isAdminEnforced": false,
  "pattern": "release-*",
  "pushAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
  "repoId": undefined,
  "repoName": "reimagined-barnacle",
  "requiredApprovingReviewCount": 2,
  "requiredStatusCheckContexts": Array [],
  "requiredStatusChecks": Array [],
  "requiresApprovingReviews": true,
  "requiresCodeOwnerReviews": false,
  "requiresCommitSignatures": true,
  "requiresConversationResolution": false,
  "requiresLinearHistory": false,
  "requiresStatusChecks": false,
  "requiresStrictStatusChecks": true,
  "restrictsPushes": false,
  "restrictsReviewDismissals": false,
  "reviewDismissalAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
}
`;

exports[`BranchProtectionRulesQuery #iterateBranchProtectionRules Pulling data out 8`] = `
Object {
  "allowsDeletions": false,
  "allowsForcePushes": false,
  "blocksCreations": false,
  "bypassForcePushAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
  "bypassPullRequestAllowances": Object {
    "apps": Array [],
    "teams": Array [
      Object {
        "__typename": "Team",
        "id": "T_kwDOBfl5Zc4AWFta",
        "name": "j1-nested-team",
      },
    ],
    "users": Array [
      Object {
        "__typename": "User",
        "email": "",
        "id": "MDQ6VXNlcjE2NTYyMDg1",
        "login": "electricgull",
      },
    ],
  },
  "creator": Object {
    "login": "electricgull",
  },
  "databaseId": 27924631,
  "dismissesStaleReviews": true,
  "isAdminEnforced": true,
  "pattern": "main",
  "pushAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
  "repoId": undefined,
  "repoName": "reimagined-barnacle",
  "requiredApprovingReviewCount": 1,
  "requiredStatusCheckContexts": Array [],
  "requiredStatusChecks": Array [],
  "requiresApprovingReviews": true,
  "requiresCodeOwnerReviews": false,
  "requiresCommitSignatures": false,
  "requiresConversationResolution": false,
  "requiresLinearHistory": false,
  "requiresStatusChecks": true,
  "requiresStrictStatusChecks": false,
  "restrictsPushes": false,
  "restrictsReviewDismissals": false,
  "reviewDismissalAllowances": Object {
    "apps": Array [],
    "teams": Array [],
    "users": Array [],
  },
}
`;

exports[`BranchProtectionRulesQuery #iterateBranchProtectionRules Pulling data out 9`] = `
Object {
  "query": "
      query (
        $repoName: String!
        $repoOwner: String!
        $maxLimit: Int!
      ) {
          repository(name: $repoName, owner: $repoOwner) {
            id
            name
            branchProtectionRules(first: $maxLimit) {
              nodes {
                
  id
  requiresLinearHistory
  requiredApprovingReviewCount
  dismissesStaleReviews
  requiresCodeOwnerReviews
  requiresCommitSignatures
  isAdminEnforced
  allowsForcePushes
  allowsDeletions
  requiresConversationResolution
  pattern
  allowsDeletions
  requiresApprovingReviews
  requiredStatusCheckContexts
  creator {
    login
  }
  databaseId
  requiresStatusChecks
  requiresStrictStatusChecks
  restrictsPushes
  restrictsReviewDismissals
  requiredStatusChecks {
    context
    app {
      id
      name
    }
  }
  blocksCreations
  bypassForcePushAllowances {
    totalCount
  }
  bypassPullRequestAllowances {
    totalCount
  }
  pushAllowances {
    totalCount
  }
  reviewDismissalAllowances {
    totalCount
  }

              }
            }
          }
          ...on Query {
    rateLimit {
      limit
      cost
      remaining
      resetAt
    }
  }
      }
  ",
  "queryVariables": Object {
    "maxLimit": 100,
    "repoName": "happy-sunshine",
    "repoOwner": "J1-Test",
  },
}
`;
